app/views/tasks/show.html.slim

```diff
@@ -4,40 +4,40 @@
       .std-heading data-bs-toggle='tooltip' data-bs-placement='bottom' title=@task.title
         i.i.fa-regular.fa-file-lines
         =< truncate(@task.title, length: 100)
-      span.task-star-rating
-        - [*1..5].each do |i|
-          - if @task.average_rating >= i
-            span.fa-solid.fa-star.overall-rating data-rating=i
-          - elsif (@task.average_rating + 0.5) >= i
-            span.fa-regular.fa-star-half-stroke.overall-rating data-rating=i
-          - else
-            span.fa-regular.fa-star.overall-rating data-rating=i
-        span
-          .rating-dropdown
-            - if current_user && policy(Rating).new?
-              .btn#rate
-                => t('.button.rate')
-                span.fa-solid.fa-caret-down
-
-              .popup-rating.shadow-sm.bg-light
-                .rating data-task-id=@task.id
-                  - if @user_rating
-                    - [*1..5].each do |i|
-                      - if @user_rating >= i
-                        span.fa-solid.fa-star data-rating=i
-                      - else
-                        span.fa-regular.fa-star data-rating=i
-                  - else
-                    - [*1..5].each do |i|
-                      span.fa-regular.fa-star data-rating=i
 
-            - else
-              div[data-bs-toggle='tooltip' title=t('.guest.disabled.tooltip') data-bs-delay=150]
-                .btn#rate
-                  => t('.button.rate')
-                  span.fa-solid.fa-caret-down
+      .d-flex.align-items-center
+        span.task-star-rating.rating-subcategories-trigger.averaged-task-ratings
+          = render('task_star_rating', rating: @task.overall_rating, category: :overall_rating)
+          .rating-subcategories-overlay
+            = render('rating_subcategories', task: @task)
+        span.mx-2
+          - if current_user && policy(Rating).new_for_task?(@task)
+            .btn.btn-light.btn-sm#ratingModalOpenButton data-bs-toggle="modal" data-bs-target="#ratingModal"
+              = t('.button.rate')
+            .modal.fade#ratingModal aria-hidden="true" data-task-id=@task.id
```

Comment:
Adding `tabindex="-1"` will ensure one can close the modal by pressing <kbd>ESC</kbd >. As a publicly-funded project, we strive for great accessibility 😁.